shared/test/test_http_cache.rb

```diff
@@ -0,0 +1,349 @@
+require 'minitest/autorun'
+require 'rack/test'
+require File.expand_path '../../../deployment', __FILE__
+require 'cdo/rack/whitelist'
+require File.expand_path('../../../cookbooks/cdo-varnish/libraries/http_cache', __FILE__)
+
+ENV['RACK_ENV'] = 'test'
+
+class HttpCacheTest < Minitest::Test
+  module RackTest
+    include Rack::Test::Methods
+    def build_rack_mock_session
+      @session ||= Rack::MockSession.new(app, 'localhost.code.org')
+    end
+
+    def mock_app
+      lambda do |env|
+        @request = Rack::Request.new(env)
+        @request_cookies = @request.cookies
+        @request_env = env
+        match = @mock_responses.select do |mock_response|
+          @request.path_info == mock_response[:url] &&
```

Comment:
This a bit dense especially with the trailing sort_by.
Could you factor into a helper-method with a self-documenting name that describes what the select predicate is selecting?